### PR TITLE
Documentation: Adds SDK guidelines

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

Adding SDK guidelines to document all of the tribal knowledge for users contributing to the SDK.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: SdkDesignGuidelines.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Public contract changes
+
+### API Review
+Public contract changes must go through an API review. Please create a github issue that explains the reason for the change and proposed design. For internal Microsoft engineers please schedule a meeting to review this with the SDK team. For external customers we will provide feedback to the github issue.
+
+### What is considered public contract?
+1. Public APIs 
+2. Preview APIs
+3. Dependencies including nuget package version
+4. Behavioral changes. For example trying to convert the default serialzier from Netwonsoft to System.Text.Json.
```
Comment (FabianMeiswinkel) [inline SdkDesignGuidelines.md:12]: NIT: typo

File: SdkDesignGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+### Request options
+Request options should not be sealed.
+
+1. This allows internal teams to extend and add custom properties to request options which they can later access in lower layers of the SDK.
+2. Request options only has public properties, so there is no concern with users extending the type.
+
+
+### Central SDK
+The V3 SDK follow the [central SDK .NET guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html) where possible. V3 SDK was released before these guidelines exisited so it does not have the same types. Consistency with the current V3 public API is more important than follow the central SDK .NET guidelines as it provides a better user experience.
```
Comment (FabianMeiswinkel) [inline SdkDesignGuidelines.md:22]: NIT: typo

**Review by FabianMeiswinkel — DISMISSED** [review]

File: SdkDesignGuidelines.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply SdkDesignGuidelines.md]: serial**iz**er from Ne**wt**onsoft

**Comment by kirankumarkolli** [comment]
Its a good start we can keep enriching it. 
Thanks for taking first cut @j82w, appreciate it.

File: SdkDesignGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+### API Review
+Public contract changes must go through an API review. Please create a github issue that explains the reason for the change and proposed design. For internal Microsoft engineers please schedule a meeting to review this with the SDK team. For external customers we will provide feedback to the github issue.
+
+### What is considered public contract?
+1. Public APIs 
+2. Preview APIs
+3. Dependencies including nuget package version
+4. Behavioral changes. For example trying to convert the default serializer from Newtonsoft to System.Text.Json. 
+
+### Request options
```
Comment (kirankumarkolli) [inline SdkDesignGuidelines.md:14]: Mockability/Unit-test-ability is one more principle.

**Review by kirankumarkolli — DISMISSED** [review]

**Review by FabianMeiswinkel — DISMISSED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**